lively.morphic/components/policy.js

```diff
@@ -572,28 +627,35 @@ export class StylePolicy {
    */
   ensureStylePoliciesInSpec (spec) {
     const klass = this.constructor;
-    const overriddenMaster = this._getOverriddenMaster(spec, this);
-    const getInlinePolicy = (name) => {
-      // we need to provide a local master that actually branches off into click, hover, breakpoints or states
-      // for EACH of the sub local policies
-      const inlinePolicy = overriddenMaster && overriddenMaster.getSplitInlinePolicyFor(name);
-      if (inlinePolicy && !inlinePolicy.isPolicy && !inlinePolicy.isComponentDescriptor) return false;
-      return inlinePolicy;
-    };
+
+    if (spec.master) {
+      if (spec.master.isPolicy || spec.master.isComponentDescriptor) {
+        this.applyConfiguration({ auto: spec.master });
+      } else {
+        this.applyConfiguration(spec.master);
+      }
+      spec = obj.dissoc(spec, ['master']);
+    }
     const ensureStylePoliciesInStandalone = (spec) => {
       return tree.mapTree(spec, (node, submorphs) => {
         if (node.props) {
           if (!node.props.name) { node.props.name = this.generateUniqueNameFor(node); }
         } else if (!node.name && node !== spec) { node.name = this.generateUniqueNameFor(node); }
-        if (node.isPolicy) return node;
-        if (node.master && node !== spec) {
-          return new klass({ ...obj.dissoc(node, ['master']), submorphs }, node.master, false);
-        }
+        if (node.isPolicy) return node.copy(); // duplicate the node to prevent in place modification or the original spec
         if (node.master) {
-          return new klass({ ...obj.dissoc(node, ['master']), submorphs }, node.master, false);
+          return new klass({ ...node, submorphs }, null);
+        }
+        // this is too weak, we need to check of there is any possible master (click, hover, auto, state, breakpoint) that
+        // may potentially provide the default value here
```

Comment:
Does this still apply?

Reply:
Well, right now this question is still not fully resolved. The way styles are calculated right now is that we check if there is a `autoMaster` and if no `autoMaster` is present (which is a fringe case to begin with) we proceed to utilize the default value of the Morph implementation. The comment raised the question if it makes sense to further dig for potentially more "appropriate" default values by taking a look at other masters such as the ones found in the breakpoints or the custom states. Since it is technically possible to construct policies that do not have an `autoMaster` (via the sidebar) I raised the question in the comment. I would be fine with adjusting the comment here, to make what I just explained more clear in the code and keep it just as a reminder for the future.

Reply:
Got it. I think it would be best if we extend the comment :slightly_smiling_face: